lib/utils/index.js

```diff
@@ -894,20 +894,25 @@ function url(public_id, options = {}) {
     resultUrl += `?${token}`;
   }
 
-  let urlAnalytics = ensureOption(options, 'urlAnalytics', true);
+  const urlAnalytics = ensureOption(options, 'urlAnalytics', ensureOption(options, 'analytics', true));
 
   if (urlAnalytics === true) {
     let {
-      sdkCode,
-      sdkSemver,
-      techVersion,
-      product
+      sdkCode: sdkCodeDefault,
+      sdkSemver: sdkSemverDefault,
+      techVersion: techVersionDefault,
+      product: productDefault
     } = getSDKVersions();
+    const sdkCode = ensureOption(options, 'sdkCode', ensureOption(options, 'sdk_code', sdkCodeDefault));
```

Comment:
it'd be more like "ensureOptionWithFallbackWithFallback", I can try to see if function currying accomplishes the same thing and is readable but for now let's merge and have that released

Reply:
it's approved from my side, let's see what @magdakwiecien says

Reply:
I see no problem in "nesting" it like this, I understood how it works quickly, IMO it's even better to use composition here instead of introduce additional helpers.